py_zipkin/logging_helper.py

```diff
@@ -158,74 +163,29 @@ def __exit__(self, _exc_type, _exc_value, _exc_traceback):
 
     def _reset_queue(self):
         self.queue = []
-        self.current_size = thrift.LIST_HEADER_SIZE
+        self.current_size = 0
 
     def add_span(self, internal_span):
-        span = internal_span.build_v1_span()
-        thrift_endpoint = thrift.create_endpoint(
-            span.endpoint.port,
-            span.endpoint.service_name,
-            span.endpoint.ipv4,
-            span.endpoint.ipv6,
-        )
-
-        thrift_annotations = thrift.annotation_list_builder(
-            span.annotations,
-            thrift_endpoint,
-        )
-
-        # Binary annotations can be set through debug messages or the
-        # set_extra_binary_annotations registry setting.
-        thrift_binary_annotations = thrift.binary_annotation_list_builder(
-            span.binary_annotations,
-            thrift_endpoint,
-        )
-
-        # Add sa binary annotation
-        if span.sa_endpoint is not None:
-            thrift_sa_endpoint = thrift.create_endpoint(
-                span.sa_endpoint.port,
-                span.sa_endpoint.service_name,
-                span.sa_endpoint.ipv4,
-                span.sa_endpoint.ipv6,
-            )
-            thrift_binary_annotations.append(thrift.create_binary_annotation(
-                key=thrift.zipkin_core.SERVER_ADDR,
-                value=thrift.SERVER_ADDR_VAL,
-                annotation_type=thrift.zipkin_core.AnnotationType.BOOL,
-                host=thrift_sa_endpoint,
-            ))
-
-        thrift_span = thrift.create_span(
-            span.id,
-            span.parent_id,
-            span.trace_id,
-            span.name,
-            thrift_annotations,
-            thrift_binary_annotations,
-            span.timestamp,
-            span.duration,
-        )
-
-        encoded_span = thrift.span_to_bytes(thrift_span)
+        encoded_span, size = self.encoder.encode_span(internal_span)
```

Comment:
Is there ever going to be a case where `size` here is not `len(encoded_span)`?

If not, i'd say there's no need to return size.

If so, then you can probably pass `size` instead of `encoded_span` into `encoder.fits`. OR you could invert it a bit and say something like `encoder.bytes_left` or something.

Overall, not a huge deal though, so feel free to ignore if you don't agree.

Reply:
i vote in favor of getting rid of the 'size' return value. i think it adds a bit of extra complexity to encode_span()